api/channel.go

```diff
@@ -1142,3 +1146,12 @@ func updateNotifyProps(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 }
+
+func GetChannelMember(channelId, userId string) (model.ChannelMember, *model.AppError) {
+	result := <-Srv.Store.Channel().GetMember(channelId, userId)
+	if result.Err != nil {
+		return model.ChannelMember{}, result.Err
```

Comment:
I don't think you need this function since I don't think you need to get the channel member again. We're already querying for all the channel members on line 582 in `api/post.go`. Just update that to not toss away the objects since we need them now and then access the one you need through that map.